include/openthread/heap.h

```diff
@@ -47,6 +47,8 @@ extern "C" {
  * @brief
  *   This module includes functions that set the external OpenThread heap.
  *
+ * @note This API is deprecated and use of it is discouraged.
```

Comment:
I believe `otHeapSetCAllocFree()` is OK to use, as it allows the platform to point OpenThread to an external heap. However, `otHeapCAlloc()` and `otHeapFree()` should **not** be used, as they allow external code to use OpenThread's internal heap.

Reply:
Ah ok, if we only want to mark those two as discouraged, I will move the note to both of those functions.  Currently they show up with no documentation on openthread.io.

Reply:
I think that makes sense.